### PR TITLE
detect duplicate state definitions, better parsing errors

### PR DESCRIPTION
**Pull request by MrChico**
Addresses #73 and #70

### CODE REVIEW

File: src/Main.hs [hunk 1]
```diff
@@ -205,17 +205,24 @@ compile :: String -> Err [Claim]
...
+  hPutStrLn stderr msg
+  exitFailure
+prettyErr contents (pn, msg) | pn == lastPos = do
+  hPutStrLn stderr msg
+  let culprit = last $ lines contents
+      line = length (lines contents) - 1
+      col  = length culprit
+  hPutStrLn stderr $ show line <> " | " <> culprit
+  hPutStrLn stderr $ unpack (Text.replicate (col + (length (show line <> " | ")) - 1) " " <> "^")
+  exitFailure
```
Comment (MrChico) [inline src/Main.hs:219]:
```suggestion
  let culprit = last $ lines contents
      line = length (lines contents) - 1
      col  = length culprit
  hPutStrLn stderr $ show line <> " | " <> culprit
  hPutStrLn stderr $ unpack (Text.replicate (col + (length (show line <> " | ")) - 1) " " <> "^")
  hPutStrLn stderr msg
  exitFailure
```

**Comment by MrChico** [comment]
Unfortunately this does not include error positions for conflicting definitions

**Comment by xwvvvvwx** [comment]
Would be cool to having a test for the multiple definitions thing, but apart from that looks great :+1: :sparkles:

**Comment by MrChico** [comment]
added test case

**Comment by MrChico** [comment]
rebased on top of #68

**Review by xwvvvvwx — APPROVED** [review]

**Pull request merged**